app/controllers/nonsipp/landorproperty/LandOrPropertyListController.scala

```diff
@@ -102,21 +102,37 @@ class LandOrPropertyListController @Inject()(
     val userAnswers = request.userAnswers
     val (status, incompleteLandOrPropertyUrl) = getLandOrPropertyTaskListStatusAndLink(userAnswers, srn)
 
+    val addresses = (1 to 5).map { index =>
+      (index.toString, Address("a", s"$index test street", None, None, "", None, "", "", LookupAddress))
+    }.toMap
+
+    val prePopAddresses = (1 to 5).map { index =>
+      (index.toString, Address("a", s"$index$index test street", None, None, "", None, "", "", LookupAddress))
+    }.toMap
+
+    // Would new validation util get rolled into TaskListStatus util?
     if (status == TaskListStatus.NotStarted) {
       Redirect(routes.LandOrPropertyHeldController.onPageLoad(srn, NormalMode))
     } else if (status == TaskListStatus.InProgress) {
       Redirect(incompleteLandOrPropertyUrl)
     } else {
-      val addresses = userAnswers.map(LandOrPropertyAddressLookupPages(srn))
+
+      // Dependant on Christo's work
+
+      // Get all indexes for a section that have atleast completed the first page
```

Comment:
Is there a need to have a Mode.PrePopulation and task list, list page and CYA pages to load with this mode?
In that case prePopAddresses would have data only if this case of pre-population mode.
To reduce the risk of affecting existing functionality.
If mode is an overkill, we could consider another way of distinguishing pre-population mode, e.g. store a boolean flag `isPrePopulation' in the request.